src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/Perf.Base64.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Text.Json.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public class Perf_Base64
+    {
+        private ArrayBufferWriter<byte> _arrayBufferWriter;
+        private byte[] _data;
+        private byte[] _dataWithNoEscaping;
+        private byte[] _dataWithEscaping;
+
+        [Params(10, 100, 1000)]
+        public int NumberOfBytes { get; set; }
+
+        [Params(true, false)]
+        public bool Formatted;
+
+        [Params(true, false)]
+        public bool SkipValidation;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _arrayBufferWriter = new ArrayBufferWriter<byte>();
+            _data = ValuesGenerator.Array<byte>(NumberOfBytes);
+
+            // Results in a number of A plus padding
+            _dataWithNoEscaping = new byte[NumberOfBytes];
+
+            // Results in a lot + and /
+            _dataWithEscaping = Enumerable.Repeat(0, NumberOfBytes)
+                .Select(i => i % 2 == 0 ? 0xFB : 0xFF)
+                .Select(i => (byte)i)
+                .ToArray();
+        }
+
+        [Benchmark]
+        public void WriteByteArrayAsBase64() => WriteByteArrayAsBase64Core(_data);
+
+        [Benchmark]
+        public void WriteByteArrayAsBase64_NoEscaping() => WriteByteArrayAsBase64Core(_dataWithNoEscaping);
+
+        [Benchmark]
+        public void WriteByteArrayAsBase64_HeavyEscaping() => WriteByteArrayAsBase64Core(_dataWithEscaping);
+
+        private void WriteByteArrayAsBase64Core(byte[] data)
+        {
+            _arrayBufferWriter.Clear();
+            using (var json = new Utf8JsonWriter(_arrayBufferWriter, new JsonWriterOptions { Indented = Formatted, SkipValidation = SkipValidation }))
+            {
+                json.WriteBase64StringValue(data);
+                json.Flush();
+            }
+        }
```

Comment:
Added with fd64d4381500dadd6fa02a1364c3cca8ca80c809

For the `Utf8JsonWriter`-benchmarks I adjusted the namespaces. Is this OK?
Otherwise there's a duplication of the class-names.

Reply:
> For the Utf8JsonWriter-benchmarks I adjusted the namespaces. Is this OK?

Unfortunately not. Namespaces are part of benchmark ID in our reporting system. We can't change namespace/type names/method names and even argument names. (if ID changes we lose history in our reporting system)

Could you please revert the namespace changes?

Reply:
I think that you can just merge both `Perf_Base64` types into one class. 

You can use `Target` property of the `GlobalSetupAttribute` to have different setups per write/read benchmarks

 ```cd
[GlobalSetup(Targets = new[] { nameof(method1), nameof(method2) })]
```

Reply:
> Could you please revert the namespace changes?

Of course. I presumed that will be the case, now it is confirmed.

> I think that you can just merge both `Perf_Base64` types into one class.

It feels strange to have benchmarks for the `Utf8JsonReader` below the `Utf8JsonWriter`-folder. That's why the new folder got introduced. I'd like to keep this one. If this is OK, so is the new namespace OK?


Thanks for the hint with GlobalSetup.

Reply:
> It feels strange to have benchmarks for the `Utf8JsonReader` below the `Utf8JsonWriter`-folder. 

Agreed.